### PR TITLE
Vellum onboarding opt-in

### PR DESCRIPTION
**Pull request by orangejenny**
@millerdev 

This is close to reusable but not quite there: the modal's content would need to be customizable (probably pull it from toggle description) and also this will likely need to support toggles that are domain-based. I'm planning to reuse this when we roll out app manager v2 and make those changes then.

Planning to deploy this on Friday. Will need to update with real blog link once that's ready. @dimagi/product let me know if you have better copy.

screen shot 2017-03-02 at 1 13 48 am

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/js/rollout_modal.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                    data: {
+                        item_list: JSON.stringify([initial_page_data("toggle_item")]),
+                        append: 1,
+                    },
+                    success: function() {
+                        window.location.reload(true);
+                    },
+                    error: function() {
+                        $modal.modal('hide');
+                        alert_user(gettext('We could not turn on the new feature. You will have the opportunity ' +
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/static/hqwebapp/js/rollout_modal.js]: 'alert_user' is not defined. (no-undef)

File: corehq/apps/toggle_ui/views.py [hunk 2]
```diff
@@ -13,8 +13,9 @@
 from corehq.apps.users.models import CouchUser
 from corehq.apps.style.decorators import use_datatables
 from corehq.toggles import all_toggles, ALL_TAGS, NAMESPACE_USER, NAMESPACE_DOMAIN
+from dimagi.utils.web import json_response
```
Comment (dimagimon) [inline corehq/apps/toggle_ui/views.py]: F401 'json_response' imported but unused

File: corehq/apps/toggle_ui/views.py [hunk 3]
```diff
@@ -13,8 +13,9 @@
...
 from toggle.models import Toggle
-from toggle.shortcuts import clear_toggle_cache, parse_toggle
+from toggle.shortcuts import clear_toggle_cache, parse_toggle, set_toggle
```
Comment (dimagimon) [inline corehq/apps/toggle_ui/views.py]: F401 'set_toggle' imported but unused

File: corehq/apps/toggle_ui/views.py [hunk 4]
```diff
@@ -14,7 +14,7 @@
 from corehq.apps.style.decorators import use_datatables
 from corehq.toggles import all_toggles, ALL_TAGS, NAMESPACE_USER, NAMESPACE_DOMAIN
 from toggle.models import Toggle
-from toggle.shortcuts import clear_toggle_cache, parse_toggle
+from toggle.shortcuts import clear_toggle_cache, parse_toggle, set_toggle
```
Comment (dimagimon) [inline corehq/apps/toggle_ui/views.py]: F401 'set_toggle' imported but unused

File: corehq/apps/app_manager/templates/app_manager/v1/form_designer.html [hunk 5]
```diff
@@ -17,6 +17,9 @@
 {% block js %}{{ block.super }}
     <script src="{% static 'moment/moment.js' %}"></script>
     <script src="{% static 'requirejs/require.js' %}"></script>
+    {% if not request|toggle_enabled:'VELLUM_BETA' %}
```
Comment (benrudolph) [inline corehq/apps/app_manager/templates/app_manager/v1/form_designer.html]: this may crash with compressor (if the super block has a compress tag)

**Review by benrudolph — COMMENTED** [review]
nice good to see we're starting to make infrastructure for this type of thing

File: corehq/apps/app_manager/templates/app_manager/v1/form_designer.html (hunk 5 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/form_designer.html]: It's safe, none of the parents compress - isn't it actually impossible to put an inheritable block inside of a compress?

Comment (benrudolph) [inline corehq/apps/app_manager/templates/app_manager/v1/form_designer.html]: at one point we used to do it for all our js blocks (from our main style base.html) however that cause so many compression errors like these we got rid of it haha

**Comment by czue** [comment]
> nice good to see we're starting to make infrastructure for this type of thing

agreed, this is the future of feature rollouts!

File: corehq/apps/app_manager/templates/app_manager/v1/form_designer.html [hunk 6]
```diff
@@ -159,3 +162,47 @@
...
 {% block column_style %}hq-flush-content{% endblock %}
+
+{% block modals %}
+    {{ block.super }}
+    {% with slug="vellum_beta" name="New Form Builder" %}
+    {% initial_page_data "toggle_item" toggle_item %}
+    {% initial_page_data "toggle_url" toggle_url %}
+    {% if not request|toggle_enabled:'VELLUM_BETA' %}
+        <!-- This will appear on page load, so don't use any animation (normally controlled by .fade) -->
+        {% blocktrans %}
```
Comment (millerdev) [inline corehq/apps/app_manager/templates/app_manager/v1/form_designer.html]: Did you intend to have this entire block translated? If you did, what do you think of moving the `{% blocktrans %}` inside `<div class="modal-content">` so there's less HTML to be processed by the translators?

**Review by millerdev — APPROVED** [review]

File: corehq/apps/app_manager/templates/app_manager/v1/form_designer.html (hunk 6 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/form_designer.html]: updated

https://github.com/dimagi/commcare-hq/pull/15207/commits/30dbc3f626bfd3939f233b7c577abea2175860e8

**Review by millerdev — APPROVED** [review]

**Comment by orangejenny** [comment]
Added blog link. Not waiting on tests; this needs to go into today's deploy (fyi @dannyroberts )

**Pull request merged**